registry/reference.go

```diff
@@ -164,13 +175,26 @@ func (r Reference) ValidateReference() error {
 	if r.Reference == "" {
 		return nil
 	}
-	if _, err := r.Digest(); err == nil {
+
+	return fmt.Errorf("%w: invalid non-tag, non-digest reference: %v", errdef.ErrInvalidReference, r.Reference)
+}
+
+// ValidateReferenceAsTag validates the reference as a tag.
+func (r Reference) ValidateReferenceAsTag() error {
+	if tagRegexp.MatchString(r.Reference) {
 		return nil
 	}
-	if !tagRegexp.MatchString(r.Reference) {
-		return fmt.Errorf("%w: invalid tag", errdef.ErrInvalidReference)
+
+	return fmt.Errorf("%w: invalid tag", errdef.ErrInvalidReference)
+}
+
+// ValidateReferenceAsDigest validates the reference as a digest.
+func (r Reference) ValidateReferenceAsDigest() (err error) {
+	if _, err = r.Digest(); err == nil {
+		return nil
 	}
-	return nil
+
+	return fmt.Errorf("%w: invalid digest; %v", errdef.ErrInvalidReference, err)
```

Comment:
same here

Reply:
I've amended all occurrences of this pattern that I introduced.